### PR TITLE
[App Service]: Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` 

### LINKED ISSUE #19098: Azure Portal CLI error when trying to disable web app auto-swap
Hi everyone!

I'm getting this error when trying to disable an auto-swap on Web App using Portal CLI.

What I'm trying to do is:
az webapp deployment slot auto-swap --name <name> --resource-group <resourcegroup> --slot <slot> --disable

**Command Name**
`az webapp deployment slot auto-swap`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
Unable to build a model: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get', DeserializationError: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get'
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1428, in _deserialize
    found_value = key_extractor(attr, attr_desc, data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1193, in rest_key_case_insensitive_extractor
    return attribute_key_case_insensitive_extractor(key, None, working_data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1222, in attribute_key_case_insensitive_extractor
    return data.get(found_key)
AttributeError: 'str' object has no attribute 'get'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 620, in body
    data = deserializer._deserialize(data_type, data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1446, in _deserialize
    raise_with_traceback(DeserializationError, msg, err)
  File "/opt/az/lib/python3.6/site-packages/msrest/exceptions.py", line 51, in raise_with_traceback
    raise error.with_traceback(exc_traceback)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1428, in _deserialize
    found_value = key_extractor(attr, attr_desc, data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1193, in rest_key_case_insensitive_extractor
    return attribute_key_case_insensitive_extractor(key, None, working_data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1222, in attribute_key_case_insensitive_extractor
    return data.get(found_key)
msrest.exceptions.DeserializationError: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 691, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/appservice/custom.py", line 2152, in config_slot_auto_swap
    return client.web_apps.update_configuration_slot(resource_group_name, webapp, site_config, slot)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/web/v2020_09_01/operations/_web_apps_operations.py", line 14737, in update_configuration_slot
    body_content = self._serialize.body(site_config, 'SiteConfigResource')
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 623, in body
    SerializationError, "Unable to build a model: "+str(err), err)
  File "/opt/az/lib/python3.6/site-packages/msrest/exceptions.py", line 51, in raise_with_traceback
    raise error.with_traceback(exc_traceback)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 620, in body
    data = deserializer._deserialize(data_type, data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1446, in _deserialize
    raise_with_traceback(DeserializationError, msg, err)
  File "/opt/az/lib/python3.6/site-packages/msrest/exceptions.py", line 51, in raise_with_traceback
    raise error.with_traceback(exc_traceback)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1428, in _deserialize
    found_value = key_extractor(attr, attr_desc, data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1193, in rest_key_case_insensitive_extractor
    return attribute_key_case_insensitive_extractor(key, None, working_data)
  File "/opt/az/lib/python3.6/site-packages/msrest/serialization.py", line 1222, in attribute_key_case_insensitive_extractor
    return data.get(found_key)
msrest.exceptions.SerializationError: Unable to build a model: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get', DeserializationError: Unable to deserialize to object: type, AttributeError: 'str' object has no attribute 'get'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az webapp deployment slot auto-swap --name {} --resource-group {} --slot {} --disable`

## Environment Summary
```
Linux-5.4.0-1055-azure-x86_64-with-debian-10.2 (Cloud Shell)
Python 3.6.10
Installer: DEB

azure-cli 2.26.0 *

Extensions:
ai-examples 0.2.5
ssh 0.1.5

```

### PR DESCRIPTION
**Pull request by parth21999**
**Description**

Fix #19098

`az webapp deployment slot auto-swap --name --resource-group --slot --disable` gives error.

Fix parameter order  in `update_configuration_slot`.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by zhoxing-ms** [comment]
Because this is a bug fix, does this change need to be written into the release note?

**Comment by zhoxing-ms** [comment]
Could you add some tests for this change?

**Comment by parth21999** [comment]
> @parth21999 Because this is a bug fix, does this change need to be written into the release note?

No, it doesn't need to be included in the release note. As per my understanding, `{}` meant that it shouldn't be included. Am I right?

**Comment by zhoxing-ms** [comment]
@parth21999  Yes. However, we usually recommend putting the bug fixes information into the release note to let users know.

By the way, why didn't we find this bug by running the tests before? If it is because of the lack of tests, could you add some relevant tests?

**Comment by parth21999** [comment]
I've added tests to cover auto-swap.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -1633,7 +1633,7 @@ def enable_local_git(cmd, resource_group_name, name, slot=None):
         client.web_apps.create_or_update_configuration(resource_group_name, name, site_config)
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:1633]: Change the if else block to use the _generic_site_operation instead

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -2171,7 +2171,7 @@ def config_slot_auto_swap(cmd, resource_group_name, webapp, slot, auto_swap_slot
     client = web_client_factory(cmd.cli_ctx)
     site_config = client.web_apps.get_configuration_slot(resource_group_name, webapp, slot)
     site_config.auto_swap_slot_name = '' if disable else (auto_swap_slot or 'production')
-    return client.web_apps.update_configuration_slot(resource_group_name, webapp, site_config, slot)
+    return client.web_apps.update_configuration_slot(resource_group_name, webapp, slot, site_config)
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2174]: Same comment here as well plus make sure to use use positional arguments here, so this doesn't cause issues in the future.

**Review by panchagnula — CHANGES_REQUESTED** [review]
a few comments to address. Thanks

**Comment by zhoxing-ms** [comment]
@parth21999 Could you resolve these conflicts by tomorrow? Otherwise, it will not catch up with the release time of this sprint, and we will plan to release it in the next sprint **(10-05)**

**Review by panchagnula — APPROVED** [review]
approved

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**